src/errors/value_exception.rs

```diff
@@ -74,13 +75,13 @@ impl PydanticCustomError {
     }
 
     #[getter(r#type)]
-    pub fn error_type(&self) -> String {
-        self.error_type.clone()
+    pub fn error_type(&self) -> &str {
+        &self.error_type
     }
 
     #[getter]
-    pub fn message_template(&self) -> String {
-        self.message_template.clone()
+    pub fn message_template(&self) -> &str {
+        &self.message_template
```

Comment:
Cool. I'm taking notes on the usage of references instead of cloning here :).

Reply:
For this `#[getter]` PyO3 will be copying either of the `&str` or `String` into a Python `str`, so not cloning here is avoiding an extra copy on the Rust side.